### PR TITLE
Feature: Add a check for TmValidatorListCollection message.

### PR DESCRIPTION
**Pull request by lwojtow**
1. Moved key constants to higher level module.
2. Test for a valid TmValidatorListCollection message.

### CODE REVIEW

File: src/tests/conformance/cmd/validators.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+#[derive(Deserialize)]
+struct ValidatorList {
+    validators: Vec<Validator>,
+}
+
+#[derive(Deserialize)]
+struct Validator {
+    validation_public_key: String,
+    manifest: String,
+}
```
Comment (ljedrz) [inline src/tests/conformance/cmd/validators.rs:75]: I think I'd prefer to have the objects defined before the test(s)

File: src/tests/conformance/mod.rs [hunk 2]
```diff
@@ -12,6 +12,14 @@ mod query;
 mod stateful;
 mod status;
 
+pub const PUBLIC_KEY_TYPES: &[u8] = &[
```
Comment (ljedrz) [inline src/tests/conformance/mod.rs:15]: haven't we added this one elsewhere recently :detective:?

Comment (ljedrz) [reply src/tests/conformance/mod.rs:15]: ah, it's just moved :+1:

File: src/tests/conformance/query/peer_shard_info.rs [hunk 3]
```diff
@@ -96,7 +90,7 @@ async fn check_relay_for_key_type(key_type: u8, relays: u32) {
 
     // Create a dummy key with the specified key type.
     let mut key = vec![key_type]; // Place the key type as the first byte.
-    key.resize(33, 0x1); // Append 32 bytes serving as a dummy public key.
+    key.resize(PUBLIC_KEY_LENGTH + 1, 0x1); // Append 32 bytes serving as a dummy public key.
```
Comment (ljedrz) [inline src/tests/conformance/query/peer_shard_info.rs:93]:
```suggestion
    key.resize(PUBLIC_KEY_LENGTH, 0x1); // Append 32 bytes serving as a dummy public key.
```

Comment (ljedrz) [reply src/tests/conformance/query/peer_shard_info.rs:93]: that, or the `const` should be `32B`

**Review by ljedrz — COMMENTED** [review]
LGTM in general, left some comments

File: src/tests/conformance/mod.rs (hunk 2 above)
Comment (lwojtow) [reply src/tests/conformance/mod.rs:15]: Yes, but now it has better name and it's moved up to be accessible in more places.

File: src/tests/conformance/query/peer_shard_info.rs (hunk 3 above)
Comment (lwojtow) [reply src/tests/conformance/query/peer_shard_info.rs:93]: Thanks. It made one test fail, but I didn't run it after changing this :-/

File: src/tests/conformance/cmd/validators.rs (hunk 1 above)
Comment (lwojtow) [reply src/tests/conformance/cmd/validators.rs:75]: Done

File: src/tests/conformance/query/peer_shard_info.rs (hunk 3 above)
Comment (lwojtow) [reply src/tests/conformance/query/peer_shard_info.rs:93]: Done

**Review by ljedrz — APPROVED** [review]
LGTM :+1:

**Pull request merged**